api/v1alpha1/inclusterippool_types.go

```diff
@@ -41,6 +41,27 @@ type InClusterIPPoolSpec struct {
 
 // InClusterIPPoolStatus defines the observed state of InClusterIPPool.
 type InClusterIPPoolStatus struct {
+	// Addresses reports the count of total, free, and used IPs in the pool.
+	// +optional
+	Addresses InClusterIPPoolStatusIPAddresses `json:"ipAddresses,omitempty"`
+}
+
+// InClusterIPPoolStatusIPAddresses contains the count of total, free, and used IPs in a pool.
+type InClusterIPPoolStatusIPAddresses struct {
+	// Total is the total number of IPs configured for the pool.
+	// Counts greater than int can contain will report as math.MaxInt.
+	// +optional
+	Total int `json:"total,omitempty"`
+
+	// Free is the count of unallocated IPs in the pool.
+	// Counts greater than int can contain will report as math.MaxInt.
+	// +optional
+	Free int `json:"free,omitempty"`
+
+	// Used is the count of allocated IPs in the pool.
+	// Counts greater than int can contain will report as math.MaxInt.
+	// +optional
+	Used int `json:"used,omitempty"`
```

Comment:
maybe we should remove `omitempty` for all of these. Otherwise `0` will not be shown, and `Free: 0` would be pretty useful.

Reply:
Ya, we tried that. Unfortunately, 0 is considered empty. Unless you know of a way to get around that, I think we're stuck with omitempty.

Reply:
Thinking more about it, what I said is probably technically not true - it's the optional setting that rejects 0. We did try different combinations of optional and omitempty and ended back with where we are.

Reply:
There is one way to make this work: turning Addresses into a pointer. I generally try to avoid pointers, but maybe it's worth it for the UX here? On the other hand, not being able to see the 0 is not that big of a deal.

Reply:
Of course, yes. Done!